redis/redis.go

```diff
@@ -131,3 +131,12 @@ func NewClient(options Options) (Client, error) {
 
 	return result, nil
 }
+
+// NewWithClient method creates a new `gokv.redis.Client` with given `redis.Client`.
+// redis related options are ignored, only Codec is applicable
+func NewWithClient(rc *redis.Client, options Options) *Client {
+	if options.Codec == nil {
+		options.Codec = DefaultOptions.Codec
+	}
+	return &Client{c: rc, codec: options.Codec}
```

Comment:
The `NewClient` currently includes a `Ping` call to check/establish a connection. Maybe we can do the same here, to minimize `NewClient` vs `NewWithClient` differences?